jmclient/jmclient/wallet.py

```diff
@@ -688,14 +696,33 @@ def select_utxos(self, mixdepth, amount, utxo_filter=None,
             assert len(i) == 2
             assert isinstance(i[0], bytes)
             assert isinstance(i[1], numbers.Integral)
-        ret = self._utxos.select_utxos(
+        utxos = self._utxos.select_utxos(
             mixdepth, amount, utxo_filter, select_fn, maxheight=maxheight)
 
-        for data in ret.values():
+        total_value = 0
+        native_utxo = None
+        for key, data in utxos.items():
+            if self._is_native_utxo(data['path']):
+                native_utxo = key
+            total_value += data['value']
             data['script'] = self.get_script_from_path(data['path'])
             if includeaddr:
                 data["address"] = self.get_address_from_path(data["path"])
-        return ret
+
+        if require_auth_address and not native_utxo:
+            # try to select more utxos, hoping for a native one
+            try:
+                return self.select_utxos(
```

Comment:
A recursive call worries me a tad here. Aren't we quite reasonably going to go into the millions ++? So we run out of stack way before that? I feel like I must have missed something. Amounts are in sats, right.

 In terms of program logic it certainly makes sense, and will certainly remove some failure cases (the main one that'll stay is the "only timelocked utxo in m0").

Reply:
No, the maximum recursive depth here is `(non-native utxos in mixdepth) + 1`. Note we're using `total_value` (of all used utxos) here, not `amount`.

Reply:
Ah I see. Yes nice trick. A user who runs out of stack space here has other problems :)